jitexecutor/jitexecutor-dmn/src/test/java/org/kie/kogito/jitexecutor/dmn/JITDMNServiceImplTest.java

```diff
@@ -162,6 +162,76 @@ void testEvaluationHitIds() throws IOException {
         Assertions.assertTrue(evaluationHitIds.contains(ruleId4));
     }
 
+    @Test
+    void testConditionalWithNestedDecisionTableFromRiskScoreEvaluation() throws IOException {
+        final String thenElementId = "_6481FF12-61B5-451C-B775-4143D9B6CD6B";
+        final String thenRuleId0 = "_D1753442-03F0-414B-94F8-6A86182DF6EB";
+        final String thenRuleId4 = "_E787BA51-E31D-449B-A432-50BE7466A15E";
+        final String elseElementId = "_2CD02CB2-6B56-45C4-B461-405E89D45633";
+        final String elseRuleId2 = "_945A5471-9F91-4751-9D96-74978F6FB12B";
+        final String elseRuleId5 = "_654BBFBC-9B84-4BD8-9D0B-13E8DD1B9F5D";
+        String decisionTableModel = getModelFromIoUtils("valid_models/DMNv1_5/RiskScore_Conditional.dmn");
+
+        Map<String, Object> context = new HashMap<>();
+        context.put("Credit Score", "Poor");
+        context.put("DTI", 33);
+        context.put("World Region", "Asia");
+        JITDMNResult dmnResult = jitdmnService.evaluateModel(decisionTableModel, context);
+
+        Assertions.assertTrue(dmnResult.getMessages().isEmpty());
+        Assertions.assertEquals(BigDecimal.valueOf(50), dmnResult.getDecisionResultByName("Risk Score").getResult());
+        List<String> evaluationHitIds = dmnResult.getEvaluationHitIds();
+        Assertions.assertNotNull(evaluationHitIds);
+        Assertions.assertEquals(3, evaluationHitIds.size());
+        Assertions.assertTrue(evaluationHitIds.contains(thenElementId));
+        Assertions.assertTrue(evaluationHitIds.contains(thenRuleId0));
+        Assertions.assertTrue(evaluationHitIds.contains(thenRuleId4));
+
+        context = new HashMap<>();
+        context.put("Credit Score", "Excellent");
+        context.put("DTI", 10);
+        context.put("World Region", "Europe");
+        dmnResult = jitdmnService.evaluateModel(decisionTableModel, context);
+
+        Assertions.assertTrue(dmnResult.getMessages().isEmpty());
+        Assertions.assertEquals(BigDecimal.valueOf(30), dmnResult.getDecisionResultByName("Risk Score").getResult());
+        evaluationHitIds = dmnResult.getEvaluationHitIds();
+        Assertions.assertNotNull(evaluationHitIds);
+        Assertions.assertEquals(3, evaluationHitIds.size());
+        Assertions.assertTrue(evaluationHitIds.contains(elseElementId));
+        Assertions.assertTrue(evaluationHitIds.contains(elseRuleId2));
+        Assertions.assertTrue(evaluationHitIds.contains(elseRuleId5));
+    }
+
+    @Test
+    void testCollectionsEvaluation() throws IOException {
```

Comment:
I would prefer a better describing name for the model - and for the test: what about something "MultipleHitRules" - or similar ?

Reply:
renamed